### DISCUSSION TITLE
OpenApi spec for Trino rest client

### DISCUSSION THREAD

**Discussion by Nexengineer** [trinodb/trino#22057] [Q&A] [answered]
Hi,

I am looking for Open api spec of Rest api present in trino to access via api. Please provide one or Point me how to create one.

Regards

**Comment by nineinchnick** [comment, marked as answer]
There's no official spec, and I'm not aware of any third-party ones. We usually don't recommend using the API directly, instead use any existing client. Why do you need the spec for?

**Comment by Nexengineer** [comment]
@nineinchnick i have a application which is exposes onprem deployment to public Internet and it requires the specs

**Reply by nineinchnick** [reply]
Why it requires the specs? Can it use an existing Trino client?

**Reply by Nexengineer** [reply]
The application is a load balancer with its own oauth2 and I am using trino gateway plugin which I want to expose via it

**Reply by nineinchnick** [reply]
What's "trino gateway plugin"? Is it related to https://github.com/trinodb/trino-gateway/? Are you writing a server or a client?

**Reply by Nexengineer** [reply]
Yes, I have Trino gateway deployed in k8s on onprem and to expose it to outside internet I have to go via that application which requires spec

**Reply by nineinchnick** [reply]
I'm afraid you have to write the spec yourself to conform to the load balancer requirements.

**Comment by findepi** [comment]
The Trino interface is documented at https://trino.io/docs/current/develop/client-protocol.html

but as @nineinchnick said, we recommend using existing clients.
